src/protocol/templates/ping1d.h.in

```diff
@@ -0,0 +1,202 @@
+/*
+* ping1d.h
+* A device API for the Blue Robotics Ping1D echosounder
+*
+*~!~!~!~!~!~!~!~!~!~!~!~!~!~!~!~!~!~!~!~!~!~!~!~!~!~!~!~!~!
+* THIS IS AN AUTOGENERATED FILE
+* DO NOT EDIT
+*~!~!~!~!~!~!~!~!~!~!~!~!~!~!~!~!~!~!~!~!~!~!~!~!~!~!~!~!~!
+*
+*/
+
+#pragma once
+
+#include <Arduino.h>
+#include <Stream.h>
+#include "pingmessage.h"
+#include "pingmessage_all.h"
+#include "ping_parser.h"
+
+class Ping1D
+{
+public:
+
+    /**
+     *  @brief Constructor
+     *  @param ser: The device I/O
+     *  @param baudrate: The device I/O baudrate
+     */
+    Ping1D(Stream& ser);
+
+    /** Destructor
+     *
+     */
+    ~Ping1D();
+
+    /**
+     *  @brief Read in data from device, return a PingMessage if available.
+     *  Data will be read in from device until there is no data left in the RX buffer,
+     *  or a valid PingMessage is successfully decoded.
+     *  Note that there may still be data available in the RX buffer for decoding when
+     *  this function returns a PingMessage.
+     *
+     *  @return: The next PingMessage from the device
+     *  @return: null if the RX buffer is empty and no PingMessage has been decoded
+     */
+    PingMessage* read();
+
+    /**
+     *  @brief Write data to device
+     *
+     *  @param data: pointer to buffer to write
+     *  @param length: buffer length to write
+     *
+     *  @return: size of data buffer written to the device
+     */
+    size_t write(uint8_t* data, uint16_t length);
+
+    /**
+     *  @brief Establish communications with the device, and initialize the update interval
+     *
+     *  @param ping_interval_ms: The interval (in milliseconds) between acoustic measurements
+     *
+     *  @return true if the device was initialized successfully
+     */
+    bool initialize(uint16_t ping_interval_ms = 50);
+
+
+    /**
+     *  @brief Wait for reciept of a message with a particular message id from device
+     *
+     *  @param id: The message id to wait for
+     *  @param timeout_ms: The timeout period to wait for a matching PingMessage to be received
+     *
+     *  @return The PingMessage received with matching id
+     *  @return null if the timeout expires and no PingMessage was received with a matching id
+     */
+    PingMessage* waitMessage(enum Ping1DNamespace::msg_ping1D_id id, uint16_t timeout_ms = 500);
+
+    /**
+     *  @brief Wait for reciept of a message with a particular message id from device
+     *
+     *  @param id: The message ID to wait for
+     *  @param timeout_ms: The timeout period to wait for a matching PingMessage to be received
+     *
+     *  @return The PingMessage received with matching id
+     *  @return null if the timeout expires and no PingMessage was received with a matching id
+     */
+    void handleMessage(PingMessage* pmsg);
```

Comment:
The doxygen doc for this class is wrong.

Reply:
This was not solved.

Reply:
lost it in the rebasing..............